public/index.js

```diff
@@ -242,33 +247,52 @@ recipeButtons.forEach((button) => {
       .catch((error) => {
         console.error("Error", error);
       });
-
-    let esc = encodeURIComponent; // declare variable  at the top ?
-    let query = Object.keys(userRecipe) // declare variable  at the top ?
-      .map((k) => esc(k) + "=" + esc(userRecipe[k]))
-      .join("&");
-    fetch(`/openai?${query}`)
+      fetch(`/openai?${createQuery(userRecipe)}`)
       .then((response) => response.json())
       .then((data) => {
-        textContent = data.text.choices[0].message.content;
-        imageUrl = data.image.data[0].url;
-        mainElement.style.backgroundImage = `url(${imageUrl})`;
-        gptResponseElement.innerHTML = `${textContent}`;
-        removeElements([headline, allergies, ...recipeButtons]);
-        displayElements([
-          userWantAnotherRecipe,
-          gptResponseElement,
-          sendRecipeToUserInboxBtn,
-        ]);
+        // CREATE TEXT PROMISE
+        const textPromise = new Promise((resolve) => {
+          textContent = data.text.choices[0].message.content;
+          resolve();
+        });
+    
+        // CREATE IMAGE PROMISE
+        const imagePromise = new Promise((resolve) => {
+          imageUrl = data.image.data[0].url;
+          resolve();
+        });
+        // WAIT FOR PROMISES TO RESOLVE
```

Comment:
This calls the promise variables and tells it to only move to .then once both promises have been resolved.